server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -403,7 +443,148 @@ private boolean isCurrentSnapshotsOnly(String[] snapshots) {
                 )
             );
         }
-        CollectionUtil.timSort(snapshotInfos);
-        return Collections.unmodifiableList(snapshotInfos);
+        return sortSnapshots(snapshotInfos, sortBy, after, size, order);
+    }
+
+    private static final Comparator<SnapshotInfo> BY_START_TIME = Comparator.comparing(SnapshotInfo::startTime)
+        .thenComparing(SnapshotInfo::snapshotId);
+
+    private static final Comparator<SnapshotInfo> BY_DURATION = Comparator.<SnapshotInfo, Long>comparing(
+        sni -> sni.endTime() - sni.startTime()
+    ).thenComparing(SnapshotInfo::snapshotId);
+
+    private static final Comparator<SnapshotInfo> BY_INDICES_COUNT = Comparator.<SnapshotInfo, Integer>comparing(
+        sni -> sni.indices().size()
+    ).thenComparing(SnapshotInfo::snapshotId);
+
+    private static final Comparator<SnapshotInfo> BY_NAME = Comparator.comparing(sni -> sni.snapshotId().getName());
+
+    private static List<SnapshotInfo> sortSnapshots(
```

Comment:
This whole method pretty wet now. I opted to keep it like this and not dry it up in some tricky way because it's only here as a place holder for a proper solution anyway, so keeping it simple seemed good enough for now.